server/src/main/java/org/eclipse/openvsx/LocalRegistryService.java

```diff
@@ -111,7 +111,9 @@ public ExtensionJson getExtension(String namespace, String extensionName, String
     public ExtensionJson getExtension(String namespace, String extensionName, String targetPlatform, String version) {
         var extVersion = findExtensionVersion(namespace, extensionName, targetPlatform, version);
         var json = toExtensionVersionJson(extVersion, targetPlatform, true, false);
+        var extension = repositories.findExtension(extensionName, namespace);
         json.downloads = getDownloads(extVersion.getExtension(), targetPlatform, extVersion.getVersion());
+        json.averageRating = extension.getAverageRating();
```

Comment:
The reason that I set `averageRating` here is because I found out in `ExtensionVersion.toExtensionJson`, the averageRating is not updated immediately according to deleting/posting new reviews with new ratings of an extension. However, the `extension.averageRating` is being updated correctly in time. That's why I put an extra statement here to update the `averageRating`.

Reply:
Ok, I see what you mean. I went through the code to find out how this could happen. The `VersionService` caches the latest `ExtensionVersion`. When a review is posted or deleted your fork doesn't evict the latest version cache, which causes `findExtensionVersion` to return a cached instance of the lastest `ExtensionVersion` with an old `Extension` instance (`averageRating` not updated).

This has been fixed in `master`. Can you sync your `master` branch and rebase the `updateReview` branch? Then the extra database call shouldn't be necessary anymore.

You can sync the `master` branch by clicking `Sync fork` in the right corner of your Github repo:
![Schermafbeelding 2023-03-31 om 11 46 01](https://user-images.githubusercontent.com/14129099/229074229-0d0a1c0b-8074-49a5-8a01-0ae91f822847.png)


After syncing your fork you can rebase following these steps:
```
git checkout master
git pull
git checkout updateReview
git rebase master
<FIX CONFLICTS & COMMIT>
git rebase --continue
git push -f origin updateReview
```

Reply:
Hello @amvanbaren , thanks for your suggestion!
I've followed these commands but seems like `git push -f origin updateReview` has caused one of the checks to be failing. It looks like one of the authors of the commits is not covered by the ECA. I'm not sure if that's gonna affect anything, or what should I do to fix this? 
Thank you again for your help :)

Reply:
Hi @yiningwang11, it looks like you merged `openvsx:master` on top of your `updateReview` branch. `yiningwang11:master` is still 70 commits behind `eclipse:master`.
![yiningwang11-master](https://user-images.githubusercontent.com/14129099/229599370-9f81c351-c1dc-4d62-8dd5-9cebd220ea6a.png)

A rebase changes the base of the branch, but the changes you've made stay on top. This article explains it in further detail: https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase. The example under the Usage heading describes what we try to do here.

Are the top 5 commits for this feature or were there more commits?
![yiningwang11-commits](https://user-images.githubusercontent.com/14129099/229604313-5d1a950d-3155-4ab1-adc7-496f6507ad57.png)

Reply:
Hello @amvanbaren , thanks for your comment! I went over the article you recommended. Is this because I should have sync my `yiningwang11:master` branch and then rebase the `updateReview` feature branch, instead of merge the `openvsx:master` on top of my `updateReview` feature branch? 
Is there anything I can do to make things right at the moment? 

The top 5 commits in the screenshot you shared were for this feature, there aren't any more commits yet. (I'm still working on adding the `@Cacheable` annotation on `LocalRegistryService.getReviews` now)

Thank you :)